src/init/features/turtles/repo.ts

```diff
@@ -0,0 +1,173 @@
+import { cloudbuildOrigin } from "../../../api";
+import { FirebaseError } from "../../../error";
+import * as gcb from "../../../gcp/cloudbuild";
+import { logger } from "../../../logger";
+import * as poller from "../../../operation-poller";
+import * as utils from "../../../utils";
+import { promptOnce } from "../../../prompt";
+
+const gcbPollerOptions: Omit<poller.OperationPollerOptions, "operationResourceName"> = {
+  apiOrigin: cloudbuildOrigin,
+  apiVersion: "v2",
+  masterTimeout: 25 * 60 * 1_000,
+  maxBackoff: 10_000,
+};
+
+function extractRepoSlugFromURI(remoteUri: string): string | undefined {
+  const match = /github.com\/(.+).git/.exec(remoteUri);
+  if (!match) {
+    return undefined;
+  }
+  return match[1];
+}
+
+function generateConnectionId(stackId: string): string {
+  return `turtles-${stackId}-conn`;
+}
+
+function generateRepositoryId(): string | undefined {
+  return `turtles-repo`;
```

Comment:
can we add a comment here that this is totally arbitrary decision (and the implication of this - e.g. this means we will maintain one repo per project per region... right?)